### PR TITLE
Tweak promotion pipeline to support rpm and qualifier

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Tweak promotion pipeline to support rpm and qualifier

### Issues Resolved
Part of #2076

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Note:
@saratvemulapalli has issued a fix to the core-plugin wrong folder: https://github.com/opensearch-project/OpenSearch/pull/3193.

If that PR get merged this PR needs some changes.

Thanks.

File: vars/promoteTarArtifacts.groovy [hunk 1]
```diff
@@ -16,78 +16,94 @@ void call(Map args = [:]) {
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
```
Comment (dblock) [inline vars/promoteTarArtifacts.groovy:16]: The name of this file should change to `promoteArtifacts` or maybe `promoteDistributions`?

**Review by dblock — COMMENTED** [review]
Just one ask for the function name change, otherwise looks good!

File: vars/promoteTarArtifacts.groovy (hunk 1 above)
Comment (peterzhuamazon) [reply vars/promoteTarArtifacts.groovy:16]: I think using distribution is inaccurate compares to artifacts.
We are moving only parts of all the distribution artifacts to production bucket.
If this one is named promoteDistribution then docker/yum should also be promoted in this same lib, which is never the case anyway.

I still think promoteAritfacts is better at this point.
Open to suggestion.

Thanks.

Comment (dblock) [reply vars/promoteTarArtifacts.groovy:16]: `promoteAritfacts` works for me, currently it's called ` promoteTarAritfacts`

File: vars/promoteTarArtifacts.groovy [hunk 2]
```diff
@@ -16,78 +16,94 @@ void call(Map args = [:]) {
...
+                        bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}",
+                        path: "releases/$pluginFullPath/",
+                        workingDir: "$prefixPath/$artifactPath/builds/$filename/core-plugins/",
+                        includePathPattern: "**/${pluginName}*"
+                    )
+                }
+            }
+            
+            // We will only publish min artifacts for Tar, ignore other distributions
+            if (distribution_local.equals('tar')) {
```
Comment (dblock) [inline vars/promoteTarArtifacts.groovy:92]: This is a bad smell. Is there a way to extract the "special case of handling tar" out of the generic "promoteArtifacts"?

Comment (peterzhuamazon) [reply vars/promoteTarArtifacts.groovy:92]: They all have the same file name conversion except the distribution, hence this is the best usage.
We will only release tar min and that will not change for now.

File: vars/promoteTarArtifacts.groovy (hunk 1 above)
Comment (peterzhuamazon) [reply vars/promoteTarArtifacts.groovy]: fixed now.

**Review by dblock — APPROVED** [review]
Approved to unblock, but the fact that we have to do `distribution_local.equals('tar'))` in `promoteArtifacts` should be refactored into something better.

**Pull request merged**